### ISSUE TITLE
0.19: Improvements to schema tags

### ISSUE THREAD

**Issue by kubukoz** [disneystreaming/smithy4s#1331]
- `CollectionTag` should be unsealed: #1819 
- `Map` schemas should have a tag: #1818
- `Optional` schemas should have a tag #1826
- `Document` should be a tag? https://github.com/disneystreaming/smithy4s/issues/1659#issuecomment-2724935204

**Comment by lewisjkl** [comment]
Should we consider this issue closed? I'm not sure I see enough value in DocumentTag for 0.19

**Comment by kubukoz** [comment]
Yeah. Document would still be amazing but I don't have a pressing usecase so no issues with deferring until later/never
